### PR TITLE
Upgrade most dependencies and migrate several tests to AssertJ

### PR DESCRIPTION
**Pull request by fbenz**
`org.junit.Assert.assertThat` was deprecated for our use case with Hamcrest matchers. Thus, I either replaced it by `org.hamcrest.MatcherAssert.assertThat` or directly migrated the test to AssertJ.

### CODE REVIEW

**Comment by fbenz** [comment]
The build is green: https://travis-ci.org/github/ScaCap/spring-auto-restdocs/builds/729441306

**Review by LEQADA — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -331,19 +331,19 @@
...
                 <artifactId>junit</artifactId>
-                <version>4.12</version>
+                <version>4.13</version>
                 <scope>test</scope>
             </dependency>
             <dependency>
                 <groupId>org.mockito</groupId>
                 <artifactId>mockito-core</artifactId>
-                <version>1.10.19</version>
+                <version>3.5.10</version>
```
Comment (jmisur) [inline pom.xml:346]: This enables strict verification by default

**Review by jmisur — APPROVED** [review]
Why don't we see Travis results in PRs anymore?

**Comment by LEQADA** [comment]
> Why don't we see Travis results in PRs anymore?

Perhaps because of this?

https://github.com/ScaCap/spring-auto-restdocs/blob/0afad4983aa4ccff5e32a9788a1799bbd47f110c/.travis.yml#L40-L42

**Comment by fbenz** [comment]
> > Why don't we see Travis results in PRs anymore?
> 
> Perhaps because of this?
> 
> https://github.com/ScaCap/spring-auto-restdocs/blob/0afad4983aa4ccff5e32a9788a1799bbd47f110c/.travis.yml#L40-L42

I don't think so. Travis is actually triggered but it's not reported here. So something in the communication between Travis CI and GitHub is broken.

File: pom.xml (hunk 1 above)
Comment (fbenz) [reply pom.xml:346]: Which is good and I was surprised that I didn't have to adjust any tests.

**Pull request merged**